core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -382,11 +385,13 @@ class DiplomacyManager() {
                             sciencefromResearchAgreement()
                     }
                     DiplomacyFlags.ProvideMilitaryUnit.name -> {
-                        // Do not unset the flag
-                        if (civInfo.cities.isEmpty() || otherCiv().cities.isEmpty())
```

Comment:
If you don't check this, you'll get a crashing bug when you try to find closest cities between the two civs.
Do not unset the flag since it's possible that soon there will be a new city for one of the two, and we don't want to have to wait an entire 40 turns to get the military unit.

Reply:
Ah, ok, that seems reasonable